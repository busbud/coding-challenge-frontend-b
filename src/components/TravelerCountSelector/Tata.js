// @flow

import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField from '@material-ui/core/TextField';
import { Add, Remove } from '@material-ui/icons';
import type { Sign } from './types';
import styles from './TravelerCountSelector.css';
import { computeTravellerCount } from './helpers';

type Props = {|
  onChange: (value: number) => void,
|};

type State = {|
  travellerCount: number,
|};

export class TravelerCountSelector extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      travellerCount: 0,
    };
  }

  onAddTraveller = (sign: Sign) => {
    const { travellerCount } = this.state;
    const { onChange } = this.props;
    const newTravellerCount = computeTravellerCount(sign, travellerCount);
    this.setState({
      travellerCount: newTravellerCount,
    });
    onChange(newTravellerCount);
  };

  render() {
    return (
      <div>
        <TextField
          id="outlined-adornment-password"
          variant="outlined"
          label="Password"
          InputProps={{
            shrink: true,
            endAdornment: (
              <InputAdornment position="end">
                <IconButton
                  aria-label="Toggle password visibility"
                  onClick={() => this.onAddTraveller('-')}
                >
                  <Remove />
                </IconButton>
              </InputAdornment>
            ),
            startAdornment: (
              <InputAdornment position="end">
                <IconButton
                  aria-label="Toggle password visibility"
                  onClick={() => this.onAddTraveller('+')}
                >
                  <Add />
                </IconButton>
              </InputAdornment>
            ),
          }}
        />
      </div>
    );
  }
}
