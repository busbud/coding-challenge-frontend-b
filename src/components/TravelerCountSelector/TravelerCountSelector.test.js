import React from 'react';

import renderer from 'react-test-renderer';
import { TravelerCountSelector } from './Tata';

describe('TravelerCountSelector', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TravelerCountSelector onChange={() => {}} label="label" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
