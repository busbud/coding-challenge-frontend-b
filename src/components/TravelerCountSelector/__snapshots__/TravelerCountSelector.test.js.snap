// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelerCountSelector renders correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root-1"
  >
    <label
      className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiInputLabel-outlined-15"
      data-shrink={true}
      htmlFor="outlined-adornment-password"
    >
      Password
    </label>
    <div
      className="MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-formControl-37 MuiInputBase-adornedStart-40 MuiOutlinedInput-adornedStart-26 MuiInputBase-adornedEnd-41 MuiOutlinedInput-adornedEnd-27"
      onClick={[Function]}
      shrink={true}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-54"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <div
        className="MuiInputAdornment-root-55 MuiInputAdornment-positionEnd-58"
      >
        <button
          aria-label="Toggle password visibility"
          className="MuiButtonBase-root-65 MuiIconButton-root-59"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-64"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-68"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-77"
          />
        </button>
      </div>
      <input
        aria-invalid={false}
        className="MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputAdornedStart-51 MuiOutlinedInput-inputAdornedStart-34 MuiInputBase-inputAdornedEnd-52 MuiOutlinedInput-inputAdornedEnd-35"
        disabled={false}
        id="outlined-adornment-password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <div
        className="MuiInputAdornment-root-55 MuiInputAdornment-positionEnd-58"
      >
        <button
          aria-label="Toggle password visibility"
          className="MuiButtonBase-root-65 MuiIconButton-root-59"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-64"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-68"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-77"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
